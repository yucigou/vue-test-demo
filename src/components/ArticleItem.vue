<template>
  <section class="item">
    <div>
      {{ item.title }}
    </div>
    <div class="author">
      {{ item.authorString }}
    </div>
  </section>
</template>
<script>
export default {
  props: {
    item: Object
  }
};
</script>

<style scoped>
section.item {
  margin: 20px;
}
.author {
  font-size: smaller;
}
</style>
