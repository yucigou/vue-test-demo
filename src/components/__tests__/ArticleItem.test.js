import { shallowMount } from "@vue/test-utils";
import ArticleItem from "../ArticleItem";
import * as data from "./ArticleItem.json";

describe("ArticleItem", () => {
  it("renders the title and authors", () => {
    const wrapper = shallowMount(ArticleItem, {
      propsData: {
        item: data
      }
    });
    expect(wrapper.text()).toContain(data.title);
    expect(wrapper.text()).toContain("Zhou P");
  });

  it("renders the list item correctly", () => {
    const wrapper = shallowMount(ArticleItem, {
      propsData: {
        item: data
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
