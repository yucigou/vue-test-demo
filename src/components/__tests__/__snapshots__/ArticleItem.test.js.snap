// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem renders the list item correctly 1`] = `
<section class="item">
  <div>
    Matrine derivate MASM protects murine MC3T3-E1 osteoblastic cells against dexamethasone-induced apoptosis via the regulation of USP14/p53.
  </div>
  <div class="author">
    Zhou P, Xia D, Wang Y, Lv H, Wang Z, Xing M, Zhao Q, Xu S.
  </div>
</section>
`;
